homeassistant/components/unifi/sensor.py

```diff
@@ -192,6 +199,66 @@ def async_device_state_value_fn(hub: UnifiHub, device: Device) -> str:
     return DEVICE_STATES[device.state]
 
 
+@callback
+def async_client_wan_monitor_latency(
+    wan: Literal["WAN", "WAN2"],
+    monitor_target: str,
+    hub: UnifiHub,
+    device: Device | None,
+) -> int | None:
+    """Retrieve the monitor target from WAN monitors."""
+
+    if device is None or device.uptime_stats is None:
+        return None
+
+    uptime_stats = device.uptime_stats
+
+    if uptime_stats_wan := uptime_stats.get(wan):
+        for item in uptime_stats_wan["monitors"]:
+            if monitor_target in item["target"]:
+                return item.get("latency_average", 0)
+
+    return None
+
+
+def make_wan_latency_sensors() -> tuple[UnifiSensorEntityDescription, ...]:
+    """Create WAN latency sensors from WAN monitor data."""
+
+    def make_wan_latency_entity_description(
+        wan: Literal["WAN", "WAN2"], name: str, monitor_target: str
+    ) -> UnifiSensorEntityDescription:
+        return UnifiSensorEntityDescription[Devices, Device](
+            key=f"{name} {wan} latency",
+            entity_category=EntityCategory.DIAGNOSTIC,
+            native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+            state_class=SensorStateClass.MEASUREMENT,
+            device_class=SensorDeviceClass.DURATION,
+            entity_registry_enabled_default=False,
+            api_handler_fn=lambda api: api.devices,
+            available_fn=async_device_available_fn,
+            device_info_fn=async_device_device_info_fn,
+            name_fn=lambda _: f"{name} {wan} latency",
+            object_fn=lambda api, obj_id: api.devices[obj_id],
+            supported_fn=lambda hub, obj_id: async_client_wan_monitor_latency(
+                wan, monitor_target, hub, hub.api.devices[obj_id]
+            )
+            is not None,
+            unique_id_fn=lambda hub,
+            obj_id: f"{name.lower}_{wan.lower}_latency-{obj_id}",
+            value_fn=partial(async_client_wan_monitor_latency, wan, monitor_target),
+        )
+
+    descriptions = []
+    for wan in ["WAN", "WAN2"]:
+        descriptions += [
+            make_wan_latency_entity_description(wan, "Microsoft", "microsoft"),
+            make_wan_latency_entity_description(wan, "Google", "google"),
+            make_wan_latency_entity_description(wan, "Cloudflare", "1.1.1.1"),
+        ]
+
+    return tuple(descriptions)
```

Comment:
I think this might also be possible to simplify to a tuple comprehension
```suggestion
    wan: Literal["WAN", "WAN2"]
    return tuple(
        make_wan_latency_entity_description(wan, name, target)
        for wan in ("WAN", "WAN2")
        for name, target in (("Microsoft", "microsoft"), ("Google", "google"), ("Cloudflare", "1.1.1.1"))
    )
```